models/articles-model.js

```diff
@@ -13,3 +13,24 @@ exports.selectArticles = (article_id) => {
       return article.rows[0];
     });
 };
+
+exports.selectAllArticles = () => {
+  return db
+    .query(
+      `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COALESCE(COUNT(comments.article_id), 0) AS comment_count FROM articles 
+      LEFT JOIN comments ON articles.article_id = comments.article_id
+      GROUP BY articles.article_id
+      ORDER BY articles.created_at DESC`
+    )
+    .then((articles) => {
+      const articlesArray = articles.rows;
+      if (articlesArray.length === 0) {
+        return Promise.reject({ status: 404, msg: "Page not found" });
+      }
+
+      for (let i = 0; i < articlesArray.length; i++) {
+        articlesArray[i].comment_count = Number(articlesArray[i].comment_count);
```

Comment:
We covered this on an NC late. No need to do this. The coalesce will make a null value 0, as you're doing a count there is no need for it. In order to get a number and not a string you can cast the count as an int and you already have an example of this that we did together.

Reply:
<img width="1219" alt="Screenshot 2024-08-26 at 13 37 44" src="https://github.com/user-attachments/assets/f92694f8-9bd4-442e-9e75-61ef5161ea06">